### DISCUSSION TITLE
Upload file and send as attacment

### DISCUSSION THREAD

**Discussion by guidoDimasi** [elsa-workflows/elsa-core#5393] [Q&A]
Coan you provide ad example workflow with an activity that wait a file to be uploaded and a sequntial activity that send the uploaded files by email

**Comment by glime-ai** [comment]
Certainly! Below is an example of an ELSA workflow that waits for a file to be uploaded and then sends the uploaded file by email:

1. **Receive File Activity**: This would be a custom activity that you create to receive an uploaded file. It would likely expose an API endpoint where files can be posted and would suspend the workflow until a file is received.

2. **Send Email Activity**: Once the file is uploaded and the workflow is resumed, you can use the built-in "Send Email" activity to send the file as an attachment. You would configure the "From", "To", "Subject", and "Body" properties, and use the received file as an attachment.

Here's a pseudo-code representation of the workflow:

```plaintext
Start
|
|--> [Receive File Activity] --(File Uploaded)--> [Send Email Activity with File Attachment]
|
End
```

To implement the "Receive File Activity", you would need to create a custom activity that can handle file uploads and suspend the workflow until a file is received. Once the file is uploaded, the activity would store the file in a location accessible to the workflow and resume the workflow execution.

The "Send Email Activity" would then pick up the file from the stored location and send it as an email attachment. You would need to configure the email activity with the necessary details such as the sender's email, recipient's email, subject, and body of the email.

Please note that the actual implementation would require you to write custom code for the "Receive File Activity" and properly configure the "Send Email" activity within the ELSA workflow designer or programmatically.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by guidoDimasi** [comment]
is it possible to receive the file with an httpendpoint that bind the request to a FormFile[] variable?
